src/test-utils/random.ts

```diff
@@ -1,10 +1,16 @@
-export const randomiseCase = (text: string): string =>
-  text
+export const randomiseCase = (text: string): string => {
+  const pseudorandomDisplacement = Math.random() * Math.PI
+  return text
     .split('')
-    .map((char) =>
-      Math.round(Math.random()) ? char.toUpperCase() : char.toLowerCase(),
-    )
+    .map((char, i) => {
+      const shouldUppercase = Math.round(
+        Math.sin(pseudorandomDisplacement + (i * Math.PI) / text.length),
+      )
+
+      return shouldUppercase ? char.toUpperCase() : char.toLowerCase()
+    })
```

Comment:
The old method turned out to be unreliable because it could just so happen that each instance in which it attempted to capitalise a char would align with a number in the given string. I've replaced it with a wave of upper/lower case shifted left by a random fraction of pi.